test/Dialect/FIRRTL/lowerXMR.mlir

```diff
@@ -692,3 +692,37 @@ firrtl.circuit "ForceRelease" {
       firrtl.ref.release_initial %c, %r_p : !firrtl.uint<1>, !firrtl.rwprobe<uint<4>>
     }
   }
+
+// -----
+// Check tracking of public output refs as sv.macro.decl and sv.macro.def
+
+// CHECK-LABEL: sv.macro.decl @ref_Top_Top_a
+// CHECK-LABEL: sv.macro.def @ref_Top_Top_a "{{[{][{]0[}][}]}}" {output_file = #hw.output_file<"ref_Top_Top.sv">, symbols = [#hw.innerNameRef<@Top::@w>]}
+// CHECK-LABEL: sv.macro.decl @ref_Top_Top_b
+// CHECK-LABEL: sv.macro.def @ref_Top_Top_b "{{[{][{]0[}][}]}}.{{[{][{]1[}][}]}}" {output_file = #hw.output_file<"ref_Top_Top.sv">, symbols = [#hw.innerNameRef<@Top::@foo>, #hw.innerNameRef<@Foo::@x>]}
```

Comment:
This should be a single substitution of a hierpath.  That way the macros are robust to hierarchy manipulation.

Reply:
Then should I remove the requirement for hierpath's size to be at least 2 (unless I add it outside the circuit op, in which case it's currently not checked at all)? Or should I do the `XMRRefOp` thing where the symbol can point to a hierpath or be a single inner-ref?

Reply:
I don't know why we have XMRRefOp separate from hierpath.  But you should dig into why hierpath op requires 2, rather than assuming it can't be changed.